### ISSUE TITLE
Missing something when trying to use "OpenIdAuthenticationMechanismDefinition"

### ISSUE THREAD

**Issue by dannymk** [fturizo/ConferenceDemo#3]
This happens during the invocation of:

```
public void login() {
      try{
         var request = (HttpServletRequest) FacesContext.getCurrentInstance().getExternalContext().getRequest();
         var response = (HttpServletResponse) FacesContext.getCurrentInstance().getExternalContext().getResponse();
         LOGGER.info("DANIEL - LoginBean.login() - before authenticate()");
         **securityContext.authenticate(request, response, new AuthenticationParameters());**

      }catch(Exception e){
         e.printStackTrace();

      }
 }

```
java.lang.NullPointerException: Cannot invoke "jakarta.security.auth.message.config.AuthConfigProvider.getServerAuthConfig(String, String, javax.security.auth.callback.CallbackHandler)" because "authConfigProvider" is null

Seems I have to register an authConfigProvider somewhere but you don't have that in your example.

**Comment by fturizo** [comment]
Hi @dannymk, the error is due to the code not having an OpenId provider configured. The configuration of the provider is defined in the `OpenIDSecurityConfiguration` class:

  @Named("oidConfig")
  @ApplicationScoped
  public class OpenIDSecurityConfiguration {
  
      private static final String ROLES_CLAIM = "/roles";
  
      @Inject
      @ConfigProperty(name = "fish.payara.demos.conference.security.openid.clientId")
      private String clientID;
  
      @Inject
      @ConfigProperty(name = "fish.payara.demos.conference.security.openid.clientSecret")
      private String clientSecret;
  
      @Inject
      @ConfigProperty(name = "fish.payara.demos.conference.security.openid.provider.uri")
      private String providerUri;
  
      @Inject
      @ConfigProperty(name = "fish.payara.demos.conference.security.openid.custom.namespace")
      private String customNamespace;

The repository doesn't contain any default provider values to prevent credentials nor private details from being exposed. You'll have to configure the application with the corresponding properties of an OpenID provider of your choosing on your end for the app to work.

**Comment by dannymk** [comment]
@fturizo  I do have that configured in another bean:

```
@OpenIdAuthenticationMechanismDefinition(
  providerURI = "https://yesserver.ours/oidc/.well-known/openid-configuration",
  clientId = "myclientid",
  clientSecret = "myclientsecret",
  redirectURI = "${baseURL}/callback",
  redirectToOriginalResource = true,
  extraParameters = "audience=20230906"
  )
  @ApplicationScoped
  @Named("oidcConfig")
  public class OidcConfig {

  }
```

I probably should mention that I am using Tomee 10.0.27, JDK 21 and jakarta.jakartaee-web-api:10.0.0

**Comment by fturizo** [comment]
Sorry, @dannymk. I tested the code in both GlassFish and Payara Server and it works correctly based on the Jakarta Security spec. I'd have to review it on detail for TomEE, so I'm unsure of why it doesn't work.

**Comment by dannymk** [comment]
I found the reason here:  https://twitter.com/arjan_tijms/status/1751297432225190075
OpenId is a Jakarta EE 10 feature.TomEE is working on Jakarta EE 10 compatibility at the moment...
    On Friday, February 2, 2024 at 05:39:19 PM EST, Fabio Turizo ***@***.***> wrote:  
 
 


Sorry, @dannymk. I tested the code in both GlassFish and Payara Server and it works correctly based on the Jakarta Security spec. I'd have to review it on detail for TomEE, so I'm unsure of why it doesn't work.
